lib/http/http.c

```diff
@@ -506,6 +507,20 @@ gotheaders(struct http_cookie * H, uint8_t * buf, size_t buflen)
 		H->res.headers[i].header = s;
 		H->res.headers[i].value = s[cpos] ? &s[cpos + 1] : &s[cpos];
 		s[cpos] = '\0';
+
+		/* Strip leading optional whitespace (OWS) from the value. */
+		j = 0;
+		while (H->res.headers[i].value[j] == ' '
+		    || H->res.headers[i].value[j] == '\t')
+			H->res.headers[i].value++;
+
+		/* Strip trailing optional whitespace (OWS) from the value. */
+		j = strlen(H->res.headers[i].value) - 1;
```

Comment:
How about
```
for (vlen = j = 0; H->res.headers[i].value[j]; j++) {
    if ((H->res.headers[i].value[j] != ' ') && (H->res.headers[i].value[j] != '\t'))
        vlen = j + 1
}
H->res.headers[i].value[vlen] = '\0';
```
?

No point scanning the string in both directions...

Reply:
Two problems: I find that a bit obscure, and `H->res.headers[i].value` isn't modifiable.  The latter one is more serious, of course.

It's true that we could get around this by doing `s[cpos + 1 + strspn_len + vlen]` but that got a bit ridiculous.  I've uploaded an alternate version, which uses the string length we calculated earlier in the function.